hamilton/plugins/jupyter_magic.py

```diff
@@ -163,95 +154,154 @@ def determine_notebook_type() -> str:
 
 @magics_class
 class HamiltonMagics(Magics):
-    """Magics to facilitate Hamilton development in Jupyter notebooks"""
+    """Magics to facilitate interactive Hamilton development in notebooks."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.builder = None
+        self.notebook_env = determine_notebook_type()
 
     @magic_arguments()  # needed on top to enable parsing
+    @argument("module_name", help="Name for the module defined in this cell.")
```

Comment:
I think we agreed on this being okay if we can set a default & then having `-m` and `--module` back.